packages/data-service/tsconfig.json

```diff
@@ -2,7 +2,12 @@
   "extends": "@mongodb-js/tsconfig-compass/tsconfig.common.json",
   "compilerOptions": {
     "allowJs": true,
-    "lib": ["ES2020"],
+    "lib": [
+      "ES2020",
+      // With the implementation of cancellable actions using AbortController, we need to include DOM for types.
+      // In nodejs, AbortController support is from 15.x and we are currently on 14.x.
+      "DOM"
```

Comment:
This should not be necessary if we don't create `AbortController`

 instances in `data-service`, you should be able to do `type AbortController = EventEmitter` and the interface should match. That way ts would still complain if we try to use browser/dom stuff in data-service which is nice.

Reply:
Added a custom type for AbortSignal and avoided using `DOM` completely. 
Unfortunately we can not use `EventEmitter` because of different interfaces.